### PR TITLE
Return OneTableErrors with code:UniqueError when values are not unique

### PR DESCRIPTION
**Pull request by rjmackay**
This was failing to properly handle errors and just returning the TransactionCancelled error

### CODE REVIEW

**Comment by rjmackay** [comment]
Ah it looks like I missed some failures with dynamo v2. I'll take a look at those soon, any pointers on that would be appreciated.

File: src/Model.js [hunk 1]
```diff
@@ -545,10 +545,10 @@ export class Model {
         try {
             await this.table.transact('write', params.transaction, params)
         } catch (err) {
-            if (err.message.indexOf('ConditionalCheckFailed') >= 0) {
+            if (err instanceof OneTableError && err.code === 'TransactionCanceledException' && err.context.err.CancellationReasons.map(r => r.Code).includes('ConditionalCheckFailed')) {
```
Comment (rjmackay) [inline src/Model.js:548]: Does this need to be more blunt to work with dynamodb v2? ie. not checking CancellationReasons

Comment (mobsense) [reply src/Model.js:548]: Sorry, I don't have immediate understanding of what the V2 equivalent would be.

I do seem to remember that V2 did not include such transaction failure errors. The GO SDK did and there was a message somewhere that said V2 would not support it in Javascript.

**Comment by rjmackay** [comment]
Updated and fixed tests

**Comment by mobsense** [comment]
Thank you.

**Pull request merged**